src/ariel-os-rt/src/lib.rs

```diff
@@ -42,7 +42,8 @@ cfg_if::cfg_if! {
 const ISR_STACKSIZE: usize =
     ariel_os_utils::usize_from_env_or!("CONFIG_ISR_STACKSIZE", 8192, "ISR stack size (in bytes)");
 
-#[link_section = ".isr_stack"]
+// SAFETY: a linker script inserts this section in RAM.
```

Comment:
```suggestion
// SAFETY: a linker script inserts this section in RAM; the machine expects … there.
```

minor: That's what happens, but what is the condition for something to be placed in that section, and how is it satisfied?

Reply:
Docs say that the section needs to match expectations, e.g., place mutable stuff in writable memory.

```
This attribute is unsafe as it allows users to place data and code into sections of memory not expecting them, such as mutable data into read-only areas.
```

ref https://doc.rust-lang.org/reference/abi.html#the-link_section-attribute